docs/architecture/psa-migration/testing.md

```diff
@@ -0,0 +1,95 @@
+Testing strategy for `MBEDTLS_USE_PSA_CRYPTO`
```

Comment:
Thanks for including a testing strategy! We've too often not given it enough thought until the project was underway.